### ISSUE TITLE
Add support for OpenAPI v3

### ISSUE THREAD

**Issue by josephgardner** [stoplightio/prism#130]
I was not able to find a similar issue requesting this, so consider it requested :)

**Comment by tbarn** [comment]
hey @josephgardner! we totally agree. we are working on OAS 3 support for the whole Stoplight platform in Q2. we are working hard to release the next version of Stoplight, which is a major platform update, and then this will be the next thing we turn our focus to.

**Comment by philsturgeon** [comment]
Hey there @tbarn, any update on this? It's Q3, and it's almost a year since OpenAPI v3 came out. :)

**Comment by tbarn** [comment]
@philsturgeon I was mostly away from my laptop this week, so I am just getting back to this. Let's talk about what it would take to upgrade it - part of it requires an update to one of the Golang libraries we are using.

**Comment by philsturgeon** [comment]
@tbarn no worries! Which library specifically, and what does it need to have done? Point me at some stuff and if I can handle the Go I'll do it, otherwise I'll try and find some people that can. We've got a bunch of go developers here at WeWork recently, and they'd be happy to get mock servers.

**Comment by pytlesk4** [comment]
@philsturgeon prism uses go-openapi/spec to load in oas files, it looks like someone has started working on it, but I am sure he would love the help. Here is the issue https://github.com/go-openapi/spec/issues/21

**Comment by philsturgeon** [comment]
Awesome news that a new v3 parser is in the works over there. I've subscribed to the issue, and once the dust has settled on that issue and an implementation has been chosen/merged/etc., I'll point some of the Go people at it to see if they can help.

**Comment by jpvajda** [comment]
Any updates on supporting OAS3 anytime soon?

**Comment by stellis** [comment]
Why not use https://github.com/getkin/kin-openapi ?

**Comment by spifd** [comment]
Hi, any update for OpenAPI v3 support in stoplight.io?

**Comment by tbarn** [comment]
There's been a lot of progress on it with other improvements. You can see the work in the [develop](https://github.com/stoplightio/prism/tree/develop) branch. We are aiming for an early release in 2019.

**Comment by tbarn** [comment]
Yes, mocking, test requests can go through it, contract testing, etc. We'll be rolling out OAS3 support feature by feature, versus doing it all at once. Near-term, we are focused on getting it into the code editor and having validation/linting for OAS3 files, which is pretty much done but not released. Then things like visual editing, mocking, contract testing, etc will come. 

TL;DR Prism standalone will have support first, then it will get added to the larger Stoplight platform.

**Comment by daKmoR** [comment]
hey, a quick update about the progress would be awesome 🤗 Don't wanna start with OAS2 if prism is going to support OAS3 any time soon 🤗

**Comment by philsturgeon** [comment]
@daKmoR  Current Status: All over it like chips on gravy. 

![image](https://user-images.githubusercontent.com/67381/54039216-accba980-41c2-11e9-8471-ff80efc9a129.png)

The team has been working incredibly hard on the fantastic new version of Prism (which is a total rewrite from v2), and we'll be poking a few brave beta testers to give it a spin shortly. We need to sort out some installation docs and other things to make sure the beta testers aren't going in completely blind.

**Comment by daKmoR** [comment]
uh awesome sign me up for chips and the beta :)

I will make sure to bring my own torch just in case :p

**Comment by francbartoli** [comment]
Great! Me too interested to the beta, thx

**Comment by ntiss** [comment]
Hi guys !
So do I. I can't wait to test this !

**Comment by brianmrock** [comment]
Thanks! I have you down for the Prism Beta group :) Be on the lookout! :)

**Comment by neuroit** [comment]
Bet'a me too, thank you!

**Comment by chischaschos** [comment]
Me too, please ❤️

**Comment by philsturgeon** [comment]
Making good progress over here! We just updated to the latest version of Graphite (#191) which means we now support OpenAPI v2.0 in both JSON and YAML, and we support $ref. Before this was merged it was just JSON and no ref support. 😅 

OpenAPI v3.0 support to Graphite is a priority in the next sprint, so then Prism will get v3 support from that. Progress! :D

**Comment by NoodlesNZ** [comment]
Please add me to the beta group too, thanks!

**Comment by LesterTheTester** [comment]
I would love to be in the beta group if possible, thank you!

**Comment by gnanderson** [comment]
Hi, Gopher here - any of the Go parts open sourced and needing help? Also happy to help test this in beta.

**Comment by philsturgeon** [comment]
@gnanderson thank you so much for offering, but v3 is entirely TypeScript! As a gopher myself I´ve found working in TypeScript to be pretty familiar, especially as somebody else went to all the trouble of setting it up. 😅 

Check out the develop branch and the open issues if you are interested in contributing.

**Comment by jrejaud** [comment]
If there are any extra spots left in the beta group, I would like one please :)

**Comment by raulsann** [comment]
> @daKmoR Current Status: All over it like chips on gravy.
> 
> ![image](https://user-images.githubusercontent.com/67381/54039216-accba980-41c2-11e9-8471-ff80efc9a129.png)
> 
> The team has been working incredibly hard on the fantastic new version of Prism (which is a total rewrite from v2), and we'll be poking a few brave beta testers to give it a spin shortly. We need to sort out some installation docs and other things to make sure the beta testers aren't going in completely blind.

Hi @daKmoR we are making a decision on what system to use this week but a requirement of our company is to use OpenAPI 3. Would you be able to send me an invite to the beta group?

Thnx!

**Comment by philsturgeon** [comment]
🥁 🥁 🥁 🥁 
Done! 
🎺 🎺 🎺 🎺 

Keep an eye on #266 for updates on alpha. As always people are welcome to much around with the develop branch but no promises on stability, don't put that code in prod, etc. 👍
